src/Proto.Cluster/Partition.cs

```diff
@@ -168,92 +156,53 @@ private void MakeUnavailable(string address)
             }
         }
 
-        private int TransferOwnership(IContext context)
+        private void MemberJoined(MemberJoinedEvent msg, IContext context)
         {
-            // Iterate through the actors in this partition and try to check if the partition
-            // PID should be in is not the current one, if so initiates a transfer to the
-            // new partition.
-            var transferredActorCount = 0;
-
-            // TODO: right now we transfer ownership on a per actor basis.
-            // this could be done in a batch
-            // ownership is also racy, new nodes should maybe forward requests to neighbours (?)
-            foreach (var (actorId, _) in _partition.ToArray())
-            {
-                var address = Cluster.MemberList.GetPartition(actorId, _kind);
-
-                if (!string.IsNullOrEmpty(address) && address != Cluster.System.ProcessRegistry.Address)
-                {
-                    transferredActorCount++;
-                    TransferOwnership(actorId, address, context);
-                }
-            }
+            Logger.LogInformation("[Partition] Kind {Kind} member joined {Address}", _kind, msg.Address);
 
-            return transferredActorCount;
-        }
+            while (!Cluster.MemberList.IsMember(msg.Address)) { }

```

Comment:
What's going on here? full spinwait for address to become member?

Reply:
I might be overly cautious here but messages to the partition actor might not necessarily be processed yet. I know that event stream subscriptions process events immediately, so it might not be needed.